### PR TITLE
Add functionality to make sure guests' rooms are unique to each guest

### LINKED ISSUE #93: Make Room unique to each Guest
Make sure that the room number is unique for each Guest, meaning 2 or more guests should not have the same room number.

### PR DESCRIPTION
**Pull request by kangzongxian**
This is because it makes sense for 2 different guests to only stay in different rooms at the same time.
closes #93

### CODE REVIEW

File: src/test/java/seedu/guest/logic/commands/AddCommandTest.java [hunk 1]
```diff
@@ -129,6 +128,11 @@ public boolean hasGuest(Guest guest) {
             throw new AssertionError("This method should not be called.");
         }
 
+        @Override
+        public boolean hasSameRoom(Guest guest) {
```
Comment (jehousoh) [inline src/test/java/seedu/guest/logic/commands/AddCommandTest.java:142]: Should we add a duplicate room test as well above?

**Review by jehousoh — CHANGES_REQUESTED** [review]
Small comment but good work beast.

File: src/test/java/seedu/guest/logic/commands/AddCommandTest.java (hunk 1 above)
Comment (kangzongxian) [reply src/test/java/seedu/guest/logic/commands/AddCommandTest.java:142]: Agreed!

**Review by jehousoh — APPROVED** [review]
lgtm!

File: src/main/java/seedu/guest/model/guest/UniqueGuestList.java [hunk 2]
```diff
@@ -63,6 +74,9 @@ public void setGuest(Guest target, Guest editedGuest) {
 
         if (!target.isSameGuest(editedGuest) && contains(editedGuest)) {
             throw new DuplicateGuestException();
+            // Only check for room equality if the guest's room has been edited
+        } else if (!target.getRoom().equals(editedGuest.getRoom()) && hasSameRoom(editedGuest)) {
```
Comment (JustinPeng13) [inline src/main/java/seedu/guest/model/guest/UniqueGuestList.java]: Can we place the comment in the `else if` block?

**Review by JustinPeng13 — CHANGES_REQUESTED** [review]
Just a formatting change!

File: src/main/java/seedu/guest/model/guest/UniqueGuestList.java (hunk 2 above)
Comment (kangzongxian) [reply src/main/java/seedu/guest/model/guest/UniqueGuestList.java]: Agreed!

**Review by JustinPeng13 — APPROVED** [review]

**Pull request merged**